src/autoscaler/eventgenerator/generator/evaluator.go

```diff
@@ -77,31 +78,38 @@ func (e *Evaluator) doEvaluate(triggerArray []*models.Trigger) {
 
 		isBreached := true
 		for _, appMetric := range appMetricList {
-			if appMetric.Value == nil {
-				e.logger.Debug("should not send trigger alarm to scaling engine because there is nil-value metric", lager.Data{"trigger": trigger, "appMetric": appMetric})
+			if appMetric.Value == "" {
```

Comment:
next time the better way is len(appMetric.Value) == 0 which covers nil and empty string.